src/main/kotlin/automaton/constructor/view/AutomatonTransitionTableView.kt

```diff
@@ -0,0 +1,129 @@
+): AutomatonTableView<TransitionTableTransitionView, String>(automaton, automatonViewContext) {
+    private val filtersCount = mutableMapOf<String, Int>()
+    init {
+        transitionsByVertices.addListener(ListChangeListener {
+            while (it.next()) {
+                if (it.wasAdded()) {
+                    val addedMap = it.addedSubList.first()
+                    filtersCount.keys.forEach { filter ->
+                        addedMap.transitions[filter] = SimpleObjectProperty(listOf())
+                    }
+                }
+            }
+        })
+        transitionsByVertices.forEach { map ->
+            filtersCount.keys.forEach { filter ->
+                map.transitions[filter] = SimpleObjectProperty(listOf())
+            }
+        }
+        sourceColumn.text = "State"
+        transitionsColumns.text = "Inputs"
+    }
+
+    override fun registerVertex(vertex: AutomatonVertex) {
+        val vertexView = AutomatonBasicVertexView(vertex)
+        controller.registerAutomatonElementView(vertexView)
+        if (vertex is BuildingBlock) {
+            vertexView.hoverableTooltip(stopManagingOnInteraction = true) {
+                Pane().apply {
+                    minWidth = this@AutomatonTransitionTableView.scene.window.width / 1.5
+                    minHeight = this@AutomatonTransitionTableView.scene.window.height / 1.5
+                    maxWidth = this@AutomatonTransitionTableView.scene.window.width / 1.5
+                    maxHeight = this@AutomatonTransitionTableView.scene.window.height / 1.5
+                    val subAutomatonView = automatonViewContext.getAutomatonView(vertex.subAutomaton)
+                    add(subAutomatonView)
+                    subAutomatonView.fitToParentSize()
+                }
+            }
+        }
+        vertexToViewMap[vertex] = vertexView
+        if (transitionsByVertices.none { it.source == vertex }) {
+            transitionsByVertices.add(TransitionMap(vertex))
+        }
+    }
+
+    override fun unregisterVertex(vertex: AutomatonVertex) {
+        transitionsByVertices.removeAll { it.source == vertex }
+        vertexToViewMap.remove(vertex)
+    }
+
+    override fun registerTransition(transition: Transition) {
+        val transitionView = TransitionTableTransitionView(transition)
+        controller.registerAutomatonElementView(transitionView)
+        transitionToViewMap[transition] = transitionView
+        transition.filtersTextBinding.addListener { _, oldValue, _ ->
+            deleteTransitionFromTable(transition, oldValue)
+            addTransitionToTable(transition)
+        }
+        addTransitionToTable(transition)
+    }
+
+    override fun unregisterTransition(transition: Transition) {
+        deleteTransitionFromTable(transition)
+        transitionToViewMap.remove(transition)
+    }
+    
+    private fun addTransitionToTable(transition: Transition) {
+        if (!filtersCount.contains(transition.filtersText)) {
```

Comment:
Попытка открыть автомат через `File` -> `Examples` приводит к падению с NPE из-за того, что этот виртуальный метод вызывается из конструктора класса-родителя до инициализации полей класса-предка. Советую не вызывать виртуальные методы из конструкторов и вообще не злоупотреблять наследованием.

```
java.lang.NullPointerException: Cannot invoke "java.util.Map.containsKey(Object)" because "this.filtersCount" is null
	at automaton.constructor.view.AutomatonTransitionTableView.addTransitionToTable(AutomatonTransitionTableView.kt:81)
	at automaton.constructor.view.AutomatonTransitionTableView.registerTransition(AutomatonTransitionTableView.kt:72)
	at automaton.constructor.view.AutomatonTableView.<init>(AutomatonTableView.kt:182)
	at automaton.constructor.view.AutomatonTransitionTableView.<init>(AutomatonTransitionTableView.kt:15)
	at automaton.constructor.view.AutomatonView.<init>(AutomatonView.kt:23)
	at automaton.constructor.view.AutomatonTabView.<init>(AutomatonTabView.kt:11)
	at automaton.constructor.view.CentralView.<init>(CentralView.kt:29)
	at automaton.constructor.view.MainWindow$special$$inlined$nonNullObjectBinding$1.invoke(Observables.kt:36)
	at automaton.constructor.view.MainWindow$special$$inlined$nonNullObjectBinding$1.invoke(Observables.kt:34)
```